### PR TITLE
File to verify distance assumption

### PR DESCRIPTION
**Pull request by tedim52**
-File to verify distance assumption with euclidean distance function
-Assumption: Low distance value indicates similar Googlers.
-Currently uses a euclidean distance function on all 7 attributes.

### CODE REVIEW

File: verifying_distance.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        random_sample(DataFrame): A table with googler pairs and their l2 distances.
+    """
+    random_sample = pd.DataFrame(columns = ["user_one_id", "user_two_id", "l2"])
+    for num in range(num):
+        user_one_id = random.randint(0, size)
+        user_two_id = random.randint(0, size)
+
+        googler_one_attributes = googler_data.iloc[user_one_id].values[1:6 + 1]
+        googler_two_attributes = googler_data.iloc[user_two_id].values[1:6 + 1]
+        l2 = l2_distance(googler_one_attributes, googler_two_attributes, 6)
```
Comment (DeanAlvarez) [inline verifying_distance.py]: you could call this distance instead of l2 to further how modular the code is

**Review by DeanAlvarez — APPROVED** [review]
LGTM

File: verifying_distance.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+
+#Normalizing data incorporated by Kayla Patterson
+from sklearn.preprocessing import MinMaxScaler
+min_max_scaler = MinMaxScaler()
+googler_data.iloc[:,0:7] = min_max_scaler.fit_transform(googler_data.iloc[:,0:7])
+
+print(googler_data.head())
+
+#Create distance function to measure l2 distance between two Googlers
```
Comment (DeanAlvarez) [inline verifying_distance.py:21]: maybe define l2 distance here?

**Comment by tedim52** [comment]
-Utilized normalize function from normalizing_data.py
-Allowed function to be taken is input in sample so any distance function we write can be sampled
-Added unit test file for distance_validation.py using unittest module
-Added unit tests for euclidean aka l2 norm distance function

**Comment by tedim52** [comment]
-Changed l2 distance function to use built in numpy l2 distance function

File: distance_validation.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+# This script is for verifying our euclidean distance assumption on
+# the 7D user attribute vector space.
+
+import math
+import random
+import numpy as np
+import pandas as pd
+import normalizing_data
+
+#Take in user attribute data as DataFrame
```
Comment (meritus) [inline distance_validation.py:11]: Typically, if you ran the linter it would say put a space before the start of comments, like above.

**Review by meritus — APPROVED** [review]
LGTM

High level, non-blocking question. Potentially it would be good to describe on the file level what everything is use for **in addition** to what it does.
